.chronus/changes/tcgc-forbid-polymorphism-flatten-2024-6-23-22-12-38.md

```diff
@@ -0,0 +1,7 @@
+---
+changeKind: feature
+packages:
+  - "@azure-tools/typespec-client-generator-core"
+---
+
+throw error when trying to flattening a model with polymorphsim
```

Comment:
```suggestion
Report error diagnostic when trying to flattening a model with polymorphism
```

do not call reporting diagnostic throw as this is something you should absolutely not do in TypeSpec libraries.